paymentsheet/src/main/java/com/stripe/android/paymentsheet/paymentdatacollection/ach/USBankAccountFormViewModel.kt

```diff
@@ -477,55 +532,80 @@ internal class USBankAccountFormViewModel @Inject internal constructor(
     }
 
     private fun updatePaymentSelection(
-        linkAccountId: String,
+        resultIdentifier: ResultIdentifier,
         bankName: String?,
         last4: String?
     ) {
         if (bankName == null || last4 == null) return
 
         val paymentSelection = createNewPaymentSelection(
+            resultIdentifier = resultIdentifier,
             last4 = last4,
             bankName = bankName,
-            linkAccountId = linkAccountId,
         )
 
         _result.tryEmit(paymentSelection)
         shouldReset = true
     }
 
     private fun createNewPaymentSelection(
+        resultIdentifier: ResultIdentifier,
         last4: String,
         bankName: String,
-        linkAccountId: String,
     ): PaymentSelection.New.USBankAccount {
         val customerRequestedSave = customerRequestedSave(
             showCheckbox = args.showCheckbox,
             saveForFutureUse = saveForFutureUse.value
         )
+
+        val paymentMethodCreateParams = when (resultIdentifier) {
+            is ResultIdentifier.PaymentMethod -> {
+                PaymentMethodCreateParams.createInstantDebits(
+                    paymentMethodId = resultIdentifier.id,
+                    requiresMandate = true,
+                    productUsage = setOf("PaymentSheet"),
+                )
+            }
+            is ResultIdentifier.Session -> {
+                PaymentMethodCreateParams.create(
+                    usBankAccount = PaymentMethodCreateParams.USBankAccount(
+                        linkAccountSessionId = resultIdentifier.id,
+                    ),
+                    billingDetails = PaymentMethod.BillingDetails(
+                        name = name.value,
+                        email = email.value,
+                        phone = phone.value,
+                        address = address.value,
+                    )
+                )
+            }
+        }
+
+        val instantDebitsInfo = (resultIdentifier as? ResultIdentifier.PaymentMethod)?.let {
+            PaymentSelection.New.USBankAccount.InstantDebitsInfo(
+                paymentMethodId = it.id,
+            )
+        }
+
+        val paymentMethodOptionsParams = if (resultIdentifier is ResultIdentifier.Session) {
+            PaymentMethodOptionsParams.USBankAccount(
+                setupFutureUsage = customerRequestedSave.setupFutureUsage
+            )
+        } else {
+            null
+        }
```

Comment:
@carlosmuvi-stripe Do we need to send anything specific here for Instant Debits?